src/python/pants/engine/internals/parametrize_test.py

```diff
@@ -126,6 +126,25 @@ def test_expand(
         key=lambda value: value[0],
     )
 
+@pytest.mark.parametrize(
+    "expected,parameters,fields",
+    [
+        ([("a@g=c", {"f": "b"})], {"g": "c"}, {"f": "b"}),
+        ([("a@f=b,g=c", {"f": "b"})], {"g": "c"}, {"f": Parametrize("b")}),
+        ([("a@f=b", {"f": "b"})], {"f": "a"}, {"f": "b"}),
+    ]
+)
+def test_expand_existing_parameters(
+    expected: list[tuple[str, dict[str, Any]]], parameters: dict[str, Any], fields: dict[str, Any | Parametrize]
+) -> None:
+    assert sorted(expected) == sorted(
```

Comment:
why sort the expected list? also, having `key` for one of the sorts but not the other looks sus to me (aren't we sorting the same things on both sides?)

Reply:
I just copy/pasted the test above, I suppose this was needed in the case where the parametrize might expand into multiple addresses. Maybe I should add a test for that as well after adressing your other comments. Make sure we expand on the non-overriden parameters.